src/catalog/postgres/pg_statistic_impl.cpp

```diff
@@ -129,4 +130,137 @@ bool PgStatisticImpl::DeleteColumnStatistics(const common::ManagedPointer<transa
+}
+
+std::unique_ptr<optimizer::ColumnStatsBase> PgStatisticImpl::CreateColumnStats(
+    common::ManagedPointer<storage::ProjectedRow> all_cols_pr, table_oid_t table_oid, col_oid_t col_oid,
+    type::TypeId type) {
+  auto num_rows = *PgStatistic::STA_NUMROWS.Get(all_cols_pr, pg_statistic_all_cols_prm_);
+  auto non_null_rows = *PgStatistic::STA_NONNULLROWS.Get(all_cols_pr, pg_statistic_all_cols_prm_);
+  auto distinct_values = *PgStatistic::STA_DISTINCTROWS.Get(all_cols_pr, pg_statistic_all_cols_prm_);
+  const auto *top_k_str = PgStatistic::STA_TOPK.Get(all_cols_pr, pg_statistic_all_cols_prm_);
+  const auto *histogram_str = PgStatistic::STA_HISTOGRAM.Get(all_cols_pr, pg_statistic_all_cols_prm_);
+
+  switch (type) {
+    case type::TypeId::BOOLEAN:
+      return CreateColumnStats<execution::sql::BoolVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                        top_k_str, histogram_str, type);
+    case type::TypeId::TINYINT:
+    case type::TypeId::SMALLINT:
+    case type::TypeId::INTEGER:
+    case type::TypeId::BIGINT:
+      return CreateColumnStats<execution::sql::Integer>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                        top_k_str, histogram_str, type);
+    case type::TypeId::REAL:
+      return CreateColumnStats<execution::sql::Real>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                     top_k_str, histogram_str, type);
+    case type::TypeId::DECIMAL:
+      return CreateColumnStats<execution::sql::DecimalVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                           top_k_str, histogram_str, type);
+    case type::TypeId::TIMESTAMP:
+      return CreateColumnStats<execution::sql::TimestampVal>(table_oid, col_oid, num_rows, non_null_rows,
+                                                             distinct_values, top_k_str, histogram_str, type);
+    case type::TypeId::DATE:
+      return CreateColumnStats<execution::sql::DateVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                        top_k_str, histogram_str, type);
+    case type::TypeId::VARCHAR:
+    case type::TypeId::VARBINARY:
+      /*
+       * TODO(Joe Koshakow) The current implementation of Histogram produces meaningless results for strings. See
+       * histogram.h for a more detailed explanation. For now we just use an empty histogram.
+       */
+      return CreateColumnStats<execution::sql::StringVal>(table_oid, col_oid, num_rows, non_null_rows, distinct_values,
+                                                          top_k_str, nullptr, type);
+    default:
+      UNREACHABLE("Invalid column type");
+  }
+}
+
+template <typename T>
+std::unique_ptr<optimizer::ColumnStatsBase> PgStatisticImpl::CreateColumnStats(
```

Comment:
@malin1993ml can you help me come up with a new name? I generally tend to overload methods when they perform roughly the same task but have different signatures, otherwise, I rename them. I'm having trouble coming up with good names for these methods that distinguish what they're doing.